### DISCUSSION TITLE
Commits made via Github REST API are not verified if running as Github app

### DISCUSSION THREAD

**Discussion by mdaniline** [renovatebot/renovate#23115] [Request Help]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.105.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

According to [the docs](https://docs.renovatebot.com/configuration-options/#platformcommit), enabling `platformCommit=true` should allow all commits made by Renovatebot running as an app to be automatically verified.

I've had a look around and this seems to have been implemented as part of https://github.com/renovatebot/renovate/issues/11537.  In this issue the decision was made to use the v3 REST API instead of GraphQL to perform commits as according to https://github.com/renovatebot/renovate/issues/11537#issuecomment-1025428526 commits made via the v3 API as well as via GraphQL will be automatically signed.

My colleagues have been in touch the Github themselves, and have had the following response:
> if you use the [createCommitOnBranch GraphQL mutation](https://docs.github.com/en/enterprise-server@3.5/graphql/reference/mutations#createcommitonbranch) these will be automatically signed if web commit signing is enabled.
> 
> If you wish to use the REST API you will need to locally create a valid GPG signature for the resulting commit and include it in the API call's parameters. The GraphQL mutation and web commit signing is a simpler option for most use cases.

I have also looked at the [v3 API docs for creating a commit](https://docs.github.com/en/rest/git/commits?apiVersion=2022-11-28#create-a-commit) and it seems that a `signature` PGP string needs to be provided in order for the commit to be considered signed, which is missing in the [Renovatebot use of the API](https://github.com/renovatebot/renovate/blob/74ad0338964e19b18be6267b5b8e618f7ffbade4/lib/modules/platform/github/index.ts#L1847).

Could I please have some clarification over whether I am missing something, or whether commit signing will not work for apps as it is currently implemented?

Cheers
Max

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
The Renovate App's own commits are all verified so something is not right about what they told you. Please find public docs confirming what they said or ask them to provide.

**Comment by mdaniline** [comment]
Thank you for you help, I will try to get an answer from Github. I haven't been able to find any publically-available docs either way (the closest I have found is https://github.com/orgs/community/discussions/50055, which is just a discussion :( ). I'll post any relevant findings for future reference.

**Reply by rarkins** [reply]
That example uses REST and is only 4 months old. 

IIRC the reason we didn't use GraphQL is because it didn't allow force push to update branches

**Reply by rarkins** [reply]
You likely need an admin to turn on web commit signing - someone had the same issue last week and resolved it this way
